vitrina/api/oauth.py

```diff
@@ -0,0 +1,144 @@
+import base64
+import math
+import os
+
+import requests
+from authlib.jose import jwt, JsonWebKey, JWTClaims
+from authlib.jose.errors import BadSignatureError
+from django.conf import settings
+from django.contrib.auth.models import AnonymousUser
+from oauth2_provider.contrib.rest_framework import TokenHasScope, OAuth2Authentication
+from oauthlib.oauth2 import TokenExpiredError
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+
+from vitrina.orgs.models import Organization
+
+Secret = str
+ClientId = str
+AccessToken = str
+
+# TODO migrate to Gravitee once its ready.
+
+class OAuthClientManagement:
+    @staticmethod
+    def _bencode(value: bytes) -> str:
+        return base64.urlsafe_b64encode(value).decode().rstrip('=')
+
+    @staticmethod
+    def generate_secret(size:int=32):
```

Comment:
Let's keep up the formatting + return type is missing

```suggestion
    def generate_secret(size: int=32):
```